### PR TITLE
Add unit tests for cmd/cli/plugin/package

### LINKED ISSUE #1032: Add unit tests to `cmd/cli/plugin/package`
(This is used to request new product features)

**Describe the feature request**

It would be nice if we could add unit tests under `cmd/cli/plugin/package`, it helps catch bugs when adding new code or updating existing logic to that go package.

**Describe alternatives you've considered**

NA

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [x] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by danniel1205**
Signed-off-by: Daniel Guo <danniel1205@gmail.com>

### What this PR does / why we need it

There are several static helper functions recently added under `cmd/cli/plugin/package` without any unit tests. This PR adds some unit tests which covers over 90% of those helper functions.

In order to make it easy for adding unit tests, this PR also refactored some functions.

### Which issue(s) this PR fixes

Fixes #1032 

### Describe testing done for PR

- Integ tests against `cmd/cli/plugin/packages`

```
Ran 8 of 8 Specs in 0.002 seconds
SUCCESS! -- 8 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestPackage (0.01s)
PASS
ok      github.com/vmware-tanzu/tanzu-framework/cmd/cli/plugin/package  0.625s

```

- Build plugin and manually try `tanzu package available get` (Result TBA)

### Release note

```release-note
none
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: cmd/cli/plugin/package/package_available_get.go [hunk 1]
```diff
@@ -113,20 +115,20 @@ func packageAvailableGet(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func getValuesSchema(cmd *cobra.Command, args []string, kc kappclient.Client) error {
-	if pkgVersion == "" {
+func getValuesSchemaForPackage(namespace, name, version string, kc kappclient.Client, writer io.Writer) error {
```
Comment (danniel1205) [inline cmd/cli/plugin/package/package_available_get.go:118]: We do not have a fake `cobra.Command` available, so replacing `cmd *cobra.Command` with `writer io.Writer` is to make it possible for unit tests to pass in other `io.Writer` (i.e., `bytes.Buffer`).

File: cmd/cli/plugin/package/package_available_get.go [hunk 2]
```diff
@@ -113,20 +115,20 @@ func packageAvailableGet(cmd *cobra.Command, args []string) error {
...
+	if version == "" {
 		return errors.New("version is required when --values-schema flag is declared. Please specify <PACKAGE-NAME>/<VERSION>")
 	}
-	pkg, pkgGetErr := kc.GetPackage(fmt.Sprintf("%s.%s", pkgName, pkgVersion), packageAvailableOp.Namespace)
+	pkg, pkgGetErr := kc.GetPackage(fmt.Sprintf("%s.%s", name, version), namespace)
 	if pkgGetErr != nil {
 		if apierrors.IsNotFound(pkgGetErr) {
-			return errors.Errorf("package '%s/%s' does not exist in the '%s' namespace", pkgName, pkgVersion, packageAvailableOp.Namespace)
+			return errors.Errorf("package '%s/%s' does not exist in the '%s' namespace", name, version, namespace)
```
Comment (danniel1205) [inline cmd/cli/plugin/package/package_available_get.go:125]: Using global variable `packageAvailableOp.Namespace` makes it is hard for unit tests. So replacing with local variable.

File: cmd/cli/plugin/package/package_available_get_test.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+		It("should return error if package does not exist", func() {
+			kappClient.GetPackageReturns(nil,
+				apierrors.NewNotFound(kapppkg.Resource("package"), "dummyPackage"))
+
+			var buf bytes.Buffer
+			err := getValuesSchemaForPackage(pkgNamespace, pkgName, pkgVersion, kappClient, &buf)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should return error if schema is empty", func() {
```
Comment (ggpaue) [inline cmd/cli/plugin/package/package_available_get_test.go:112]: @danniel1205  thanks for adding the unit test, I have an open PR which not return error if values schema is empty in this case, instead it will return warning msg:
https://github.com/vmware-tanzu/tanzu-framework/pull/932
maybe we can merge these 2 PR together and change this unit test case as well?

Comment (danniel1205) [reply cmd/cli/plugin/package/package_available_get_test.go:112]: Adding hold label to this PR. Let's get #932 merged first.

**Comment by danniel1205** [comment]
close this pr, duplicate with https://github.com/vmware-tanzu/tanzu-framework/pull/1120
